src/appengine/libs/issue_management/jira/issue_tracker_manager.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Issue tracker manager functions."""
+from __future__ import absolute_import
+
+from builtins import object
+import jira
+import datetime
+import json
+
+from libs.issue_management import issue_tracker_policy
+from datastore import data_handler
+from config import db_config
+
+
+class IssueTrackerManager(object):
+    """Issue tracker manager."""
+
+    def __init__(self, project_name):
+        """"Construct an issue tracker manager instance based on parameters."""
+        self._client = None
+        self.project_name = project_name
+
+    @property
+    def client(self):
+        """HTTP Client."""
+        if self._client is None:
+            self._client = self._create_client()
+
+        return self._client
+
+    def _create_client(self):
+        """Return a client object for querying the issue tracker."""
+        # Pull credentials old-style, using native client id and json.
+        config = db_config.get()
+        credentials = json.loads(config.jira_credentials)
+        jira_url = config.jira_url
+        jira_client = jira.JIRA(jira_url, auth=(
+            credentials['username'], credentials['password']))
+        return jira_client
+
+    def save(self, issue):
+        """Save an issue."""
+        self._update(issue)
+
+    def _create(self, job_type):
+        """Create an issue."""
+        jira_project = data_handler.get_value_from_job_definition(
```

Comment:
Any reason that a separate environment variable is being used here for the jira project? We already define PROJECT_NAME in some job definitions for the monorail project name, and I believe it would be set in __init__ above.

Reply:
Good call; using PROJECT_NAME will also let me avoid having to call `data_handler.get_value_from_job_definition` and I can revert the `new_issue` call to its old signature.